sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/KafkaBundle.java

```diff
@@ -60,10 +62,12 @@
 
   private final List<MessageListener<?, ?>> messageListeners = new ArrayList<>();
   private final List<ThreadedMessageListener<?, ?>> threadedMessageListeners = new ArrayList<>();
-  private final List<KafkaMessageProducer<?, ?>> messageProducers = new ArrayList<>();
+  private final Map<String, KafkaMessageProducer<?, ?>> messageProducers = new HashMap<>();
```

Comment:
Can we stay with `List` here? It seems like the grouping by `clientId` is not needed.

Reply:
I changed that to be able to remove metrics when producers are stoped.
see line 458

Reply:
Ah … I looked for the usage but overlooked the `k`.